spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewritePositionDeleteFilesSparkAction.java

```diff
@@ -215,7 +215,9 @@ private ExecutorService rewriteService() {
   }
 
   private RewritePositionDeletesCommitManager commitManager() {
-    return new RewritePositionDeletesCommitManager(table);
+    Map<String, String> summary = Maps.newHashMap();
+    summary.put("spark.app.id", spark().sparkContext().applicationId());
```

Comment:
other idea: what about setting the app ID in `EnvironmentContext` and then later just retrieving all properties via `EnvironmentContext.get()`?. This would also include the Iceberg and Spark version

Reply:
It's already set for `SparkCatalog`. If we go down this path, can we put all of `EnvironmentContext.get()` into summary in `SnapshotProducer`?

Reply:
I don't know yet, I'd have to think more about potential implications of this. I'll try and find some time this week to review this

Reply:
@nastra do you get a chance to review this idea?

Reply:
@manuzhang sorry, due to other priorities I did not have time yet to look at this, but I'll try to this week